plutus-tx-plugin/test/IsData/deconstructData.pir.golden

```diff
@@ -363,25 +405,113 @@
       (lam
         d
         (con data)
-        [
+        (let
+          (nonrec)
+          (termbind (strict) (vardecl d (con data)) d)
+          (termbind
+            (nonstrict)
+            (vardecl
+              tup [ [ (con pair) (con integer) ] [ (con list) (con data) ] ]
+            )
+            [ unsafeDataAsConstr d ]
+          )
           [
             [
               [
                 [
                   [
-                    [ { chooseData (fun Unit [ Maybe (con integer) ]) } d ]
-                    (lam ds Unit { Nothing (con integer) })
+                    [
+                      [ { chooseData (fun Unit [ Maybe (con integer) ]) } d ]
+                      (lam
+                        ds
+                        Unit
+                        (let
+                          (nonrec)
+                          (termbind
+                            (strict)
+                            (vardecl ds (con integer))
```

Comment:
lots of dead stuff in here... we would be able to get rid of it with the purity change to builtins, but still